### PR TITLE
Fix bf16 tests using torch as reference value

### PR DESCRIPTION
**Pull request by ignaciosica**
Enables much stricter tolerance (same as for other floats) now in binary operations.
Makes it easier adding other types only supported in torch in the future.

`.float().numpy()` float casting before numpy call is required.

### CODE REVIEW

File: test/test_dtype_alu.py [hunk 1]
```diff
@@ -60,24 +64,28 @@ class ht:
...
 def universal_test(a, b, dtype, op):
-  if not isinstance(op, tuple): op = (op, op)
+  if not isinstance(op, tuple): op = (op, op, op)
   tensor_value = (op[0](Tensor([a], dtype=dtype), Tensor([b], dtype=dtype))).numpy()
-  numpy_value = op[1](np.array([a]).astype(_to_np_dtype(dtype)), np.array([b]).astype(_to_np_dtype(dtype)))
-  if dtype is dtypes.bfloat16: np.testing.assert_allclose(tensor_value, numpy_value, atol=1e-3, rtol=1e-2)
-  elif dtype in dtypes_float: np.testing.assert_allclose(tensor_value, numpy_value, atol=1e-10)
-  else: np.testing.assert_equal(tensor_value, numpy_value)
+  if dtype in torch_dtypes_map.keys():
+    reference_value = op[2](torch.tensor([a]).to(torch_dtypes_map[dtype]), torch.tensor([b]).to(torch_dtypes_map[dtype])).float().numpy()
```
Comment (chenyuxyz) [inline test/test_dtype_alu.py:70]: use `.item()` to avoid the `float()` conversion should work on torch tensor

File: test/test_dtype_alu.py [hunk 2]
```diff
@@ -60,24 +64,28 @@ class ht:
   bool = strat.booleans()
 
 def universal_test(a, b, dtype, op):
-  if not isinstance(op, tuple): op = (op, op)
+  if not isinstance(op, tuple): op = (op, op, op)
```
Comment (chenyuxyz) [inline test/test_dtype_alu.py:67]: don't really like this change, i think just using torch for bfloat16 is simpler

Comment (ignaciosica) [reply test/test_dtype_alu.py:67]: Ok, will refactor

File: test/test_dtype_alu.py (hunk 1 above)
Comment (ignaciosica) [reply test/test_dtype_alu.py]: Thanks

**Comment by ignaciosica** [comment]
Making draft as I don't like the copy paste feel of `universal_test_unary_torch`. Will work on it tommorrow. Also, one run still failed in [CI](https://github.com/tinygrad/tinygrad/actions/runs/11451286748/job/31860245877#step:22:183) in a previous run.

**Pull request closed without merging**